examples/tts/speechllm/megatron_t5_speechllm_dpo.py

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import torch.multiprocessing as mp
+from omegaconf.omegaconf import OmegaConf, open_dict
+
+from nemo.collections.tts.models.speechllm.megatron_t5_speechllm_model import MegatronT5SpeechLMModel_DPO
+from nemo.collections.nlp.parts.nlp_overrides import NLPSaveRestoreConnector
+from nemo.collections.nlp.parts.megatron_trainer_builder import MegatronTrainerBuilder
+from nemo.core.config import hydra_runner
+from nemo.utils import logging
+from nemo.utils.exp_manager import exp_manager
+import json
+import os
+import copy
+
+def read_records(manifest_path):
+    with open(manifest_path, 'r') as f:
+        lines = f.readlines()
+        records = []
+        for line in lines:
+            records.append(json.loads(line.strip()))
+    return records
+
+def write_records(fp, records):
+    with open(fp, "w") as f:
+        for record in records:
+            f.write(json.dumps(record) + "\n")
+        print(f"Wrote {len(records)} records to: {fp}")
```

Comment:
it seems these read/write manifest functions are not differing from `read_manifest` and `write_manifest` as defined in `nemo/collections/asr/parts/utils/manifest_utils.py`. It would be great to reuse the existing functions.

Reply:
Thanks! Resolved.